docs/addons/app-mesh.md

```diff
@@ -21,9 +21,67 @@ new EksBlueprint(app, 'my-stack-name', addOns, [], {
 });
 ```
 
+## App Mesh Sidecar Injection
+
+You can configure certain namespaces for automatic injection of App Mesh sidecar (Envoy) proxy. This will enable handling cross-cutting aspects such as service to service communication, resiliency patterns (Circuit breaker/retries) as well handle ingress and egress for the workloads running in the namespace.
+
+Here is an example of a team with a namespace configured for automatic sidecar injection:
+
+```typescript
+export class TeamBurnhamSetup extends ApplicationTeam {
+    constructor(scope: Construct) {
+        super({
+            name: "burnham",
+            users: getUserArns(scope, "team-burnham.users"),
+            namespaceAnnotations: {
+                "appmesh.k8s.aws/sidecarInjectorWebhook": "enabled"
+            }
+        });
+    }
+}
+```
+## Tracing Integration
+
+App Mesh integrates with a number of tracing providers for distributed tracing support. At the moment it supports AWS X-Ray, Jaeger and Datadog providers. 
+X-Ray integration at present requires either a managed node group or a self-managed auto-scaling group backed by EC2. Fargate is not supported. 
+
+Enabling integration:
+
+```typescript
+const appMeshAddOn = new ssp.AppMeshAddOn({enableTracing: true, tracingProvider: "x-ray"}),
```

Comment:
should make X-Ray an enum. We can add support for Jaeger and Datadog over time.

Reply:
yep, makes sense. will address.

Reply:
since it is a string enum and I need string values, I am going to just restrict the type. It works for switch as well and less verbose.